mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala

```diff
@@ -141,8 +141,10 @@ class CrossValidator @Since("1.2.0") (@Since("1.4.0") override val uid: String)
       Some(Array.fill($(numFolds))(Array.fill[Model[_]](epm.length)(null)))
     } else None
 
+    val inputRDD = dataset.toDF.rdd
+    inputRDD.persist()
```

Comment:
> Persisting intermediate results is not always good

Kind request, Can you explain a case for this? and can this be solved by changing the storage level?
A case when persisting intermediate result would be inefficient is, when dataset is larger than memory.

Reply:
Persisting takes time and resources. If it's only used once, it is likely to make things slower.
Even where it might be reused a few times, it may not be worth it; kind of depends on what the usages are. By default, I would not persist something unless there's a pretty clear argument for it, like a benchmark or obvious reason it's slow (used in a loop, etc)

Reply:
I think, here persist is required? Since training and testing datasets are dependent on them and in further code, these datasets have been used repeatedly.

Reply:
This is up to the caller though. At least you'd have to check if it's already persisted. We add these internal persist calls for important partial results though, not user inputs, in general, and only if the user input was cached.

Reply:
Yes, persist is not required. Training and testing datasets are already persisted.
https://github.com/apache/spark/blob/80fbc382a60973db21367a922a0fb797e5ab382d/mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala#L147-L148